src/LrtSquare.sol

```diff
@@ -413,6 +501,23 @@ contract LrtSquare is
         return IERC20Metadata(erc20).decimals();
     }
 
+    function _checkPercentagesInVault() internal view {
+        uint256 len = tokens.length;
+        uint64[] memory percentagesInVault = new uint64[](len);
+        uint256 vaultTotalValue = getVaultTokenValuesInEth(totalSupply());
+
+        if(vaultTotalValue == 0) return;
+
+        for (uint256 i = 0; i < len; ) {
+            uint256 valueOfTokenInVault = getAvsTokenTotalValuesInEth(tokens[i]);
+            percentagesInVault[i] = SafeCast.toUint64((valueOfTokenInVault * HUNDRED_PERCENT_LIMIT) / vaultTotalValue);
```

Comment:
Having a public view function which returns the position weight would be useful

`getPositionWeight(address token)`?

Reply:
Done